src/components/PortfolioBalancesProvider/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { useWeb3React } from '@web3-react/core'
+import { usePortfolioBalancesLazyQuery, usePortfolioBalancesQuery } from 'graphql/data/__generated__/types-and-hooks'
+import { GQL_MAINNET_CHAINS } from 'graphql/data/util'
+import usePrevious from 'hooks/usePrevious'
+import { createContext, ReactNode, useContext, useEffect, useMemo, useState } from 'react'
+import noop from 'utils/noop'
+
+import { usePendingActivity } from '../AccountDrawer/MiniPortfolio/Activity/hooks'
+
+const BalancesContext = createContext<{
+  query?: ReturnType<typeof usePortfolioBalancesQuery>
+  fetch: () => void
+  stale: boolean
+}>({} as any)
+
+/** Returns true if the number of pending activities has decreased */
+function useHasUpdatedTx() {
+  const { pendingActivityCount } = usePendingActivity()
+  const prevPendingActivityCount = usePrevious(pendingActivityCount)
+
+  return !!prevPendingActivityCount && pendingActivityCount < prevPendingActivityCount
+}
+
+function useHasAccountChanged() {
+  const { account } = useWeb3React()
+  const prevAccount = usePrevious(account)
+
+  return !!prevAccount && account !== prevAccount
+}
```

Comment:
Can you make a new `src/hooks/useHasChanged` hook that wraps `usePrevious` and re-use it between these two?

Reply:
the !== vs < between these two kinda hinders reusing logic here

Reply:
why not just use !==?

Reply:
you don't want to trigger a balance refetch if the number of pending transactions has gone _up_, only down (meaning something finalized on chain)